services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/resource_tracker_process_messages.py

```diff
@@ -160,12 +162,21 @@ async def _process_heartbeat_event(
             update_credit_transaction
         )
         # Publish wallet total credits to RabbitMQ
-        await sum_credit_transactions_and_publish_to_rabbitmq(
+        wallet_total_credits = await sum_credit_transactions_and_publish_to_rabbitmq(
             resource_tracker_repo,
             rabbitmq_client,
             running_service.product_name,
             running_service.wallet_id,
         )
+        if wallet_total_credits.available_osparc_credits < CreditsLimit.ZERO:
```

Comment:
Q: Think I have a good question.
As a user I will expect the following, if I hit 0 credits, I want them to be 0, not -0.12 nor -1.1.
How do we guarantee that we never go below a certain "limit"?

And what are your thoughts about this @sanderegg @pcrespov ?

Reply:
A: I think we do not know yet the final answer for this because POs want to allow users to go to negative numbers (just block the project, and after users pay we will unblock this project) Therefore I would leave it like this for now. If you have this request we can on the front end just show 0 when there is something less than 0.

I would not now artificially modify the "closing" transaction as I think the next steps will be that we will allow users to go to negative numbers. Of course, your question is valid, it will be just moved from 0 to for example -200. Still, I think maybe we do not need to add this complexity on the backend and just show the final nice number on the frontend. But I will not deal with that in this PR.

Reply:
To make it clear when I say "0" I mean a set amount, even if you don't want to go below "10" credits how would we handle this to avoid that the user has a balance lower than what was expected?

Reply:
I think the action/statement is not I don't want to go below 10. The statement is if I go under 10 then I want to pop up with for example 100 dollars.

Reply:
you might want to rename that enum from ZERO to something like MIN_CREDITS or so ?
ZERO might be a negative value right?